### PR TITLE
[ANALYTICS - FLOW] Save a snapshot before switching ending story

### PR DESCRIPTION
**Pull request by assouktim**

### CODE REVIEW

File: bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt [hunk 1]
```diff
@@ -247,63 +244,62 @@ internal object UserTimelineMongoDAO : UserTimelineDAO, UserReportDAO, DialogRep
             logger.debug { "dialog saved $userTimeline" }
         }
 
-        executor.executeBlocking {
```
Comment (vsct-jburet) [inline bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:250]: This is not a good idea to remove this executeBlocking block. Why doing this ?

Comment (assouktim) [reply bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:200]: Due to data state synchronization issue in coroutines.

Now it's fixed. I implemented the following solution: https://kotlinlang.org/docs/shared-mutable-state-and-concurrency.html#thread-confinement-coarse-grained
I

Comment (vsct-jburet) [reply bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:200]: I still don't like the PR.. I suggest we start a call about this :)

Comment (assouktim) [reply bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:200]: Yes, I'm available for the rest of the week. We can talk about this RP.
What about you?

Comment (vsct-jburet) [reply bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:200]: I'v sent a message on gitter ;)

Comment (assouktim) [reply bot/storage-mongo/src/main/kotlin/UserTimelineMongoDAO.kt:206]: Following our discussion, the code has been updated

**Review by vsct-jburet — APPROVED** [review]

**Pull request merged**